src/modals/confirmation-modal/internal-operations-confirmation/internal-operations-confirmation.tsx

```diff
@@ -0,0 +1,132 @@
+import { ButtonLargeSecondary } from '../../../components/button/button-large/button-large-secondary/button-large-secondary';
+import { Divider } from '../../../components/divider/divider';
+import { ModalButtonsContainer } from '../../../components/modal-buttons-container/modal-buttons-container';
+import { ScreenContainer } from '../../../components/screen-container/screen-container';
+import { StacksEnum } from '../../../navigator/enums/stacks.enum';
+import { useNavigation } from '../../../navigator/hooks/use-navigation.hook';
+import { useShelter } from '../../../shelter/use-shelter.hook';
+import { loadEstimationsActions } from '../../../store/wallet/wallet-actions';
+import { useEstimationsSelector } from '../../../store/wallet/wallet-selectors';
+import { formatSize } from '../../../styles/format-size';
+import { isDefined } from '../../../utils/is-defined';
+import { tzToMutez } from '../../../utils/tezos.util';
+import { ConfirmationModalParams } from '../confirmation-modal.params';
+import { FeeFormInput } from './fee-form-input/fee-form-input';
+import { FeeFormInputValues } from './fee-form-input/fee-form-input.form';
+import { useFeeForm } from './fee-form-input/use-fee-form.hook';
+import { useInternalOperationsConfirmationStyles } from './internal-operations-confirmation.styles';
+import { OperationsPreview } from './operations-preview/operations-preview';
+
+type Props = Omit<ConfirmationModalParams, 'type'>;
+
+export const InternalOperationsConfirmation: FC<Props> = ({ sender, opParams }) => {
+  const styles = useInternalOperationsConfirmationStyles();
+  const dispatch = useDispatch();
+  const { send } = useShelter();
+  const { goBack, navigate } = useNavigation();
+
+  const estimations = useEstimationsSelector();
+  const { basicFees, estimationWasSuccessful, formValidationSchema, formInitialValues } = useFeeForm(estimations.data);
+
+  useEffect(() => void dispatch(loadEstimationsActions.submit({ sender, opParams })), []);
+
+  const handleSubmit = ({ gasFee, storageFee }: FeeFormInputValues) => {
+    if (isDefined(gasFee) && isDefined(storageFee)) {
+      let params: WalletParamsWithKind[] = opParams;
+
+      if (estimationWasSuccessful) {
+        const rawAddGasFee = tzToMutez(gasFee.minus(basicFees.gasFee), 6);
+        const rawAddStorageFee = tzToMutez(storageFee.minus(basicFees.storageFee), 6);
+
+        const rawAddGasFeePerOp = rawAddGasFee.div(opParams.length).integerValue();
+        const rawAddStorageFeePerOp = rawAddStorageFee.div(opParams.length).integerValue();
+
+        params = opParams.map((param, index) => {
+          const { totalCost, storageLimit } = estimations.data[index];
+
+          return {
+            ...param,
+            fee: new BigNumber(totalCost)
+              .plus(rawAddGasFeePerOp)
+              .plus(index === opParams.length - 1 ? rawAddGasFee.mod(opParams.length) : 0),
+            storage_limit: new BigNumber(storageLimit)
+              .plus(rawAddStorageFeePerOp)
```

Comment:
We can't add `storageLimit` and `storageFee`. `storageLimit` is natural number related to one operation. `storageFee` is estimated and read-only value per all operations.